crates/cargo-wdk/src/actions/package/mod.rs

```diff
@@ -408,3 +430,31 @@ impl<'a> PackageAction<'a> {
         Ok(())
     }
 }
+
+/// # Errors
+/// Returns
+/// * `CpuArchitecture`
+/// * `RustupToolChainNotFound` error when `RUSTUP_TOOLCHAIN` environment
+///   variable is not available
+fn detect_arch_from_rustup_toolchain() -> Result<CpuArchitecture, PackageActionError> {
```

Comment:
Since this method contains side effects, we should make this method unit testable. I would have use an env provider for this.